File: JAVA/zadanie_3.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String[] numbers = scanner.nextLine().split(" ");
        for (int i = 0; i < numbers.length; i++) {
            int sum = 0, product = 1;
            for (char c : numbers[i].toCharArray()) {
                int digit = c - '0';
                sum += digit;
                product *= digit;
            }
            if (sum < product) {
                System.out.print(i + " ");
            }
        }
        scanner.close();
    }
}

File: C#/zadanie_3.cs
using System;

class Program
{
    static void Main()
    {
        string input = Console.ReadLine();
        string[] numbers = input.Split();
        for (int i = 0; i < numbers.Length; i++)
        {
            int sum = 0, product = 1;
            foreach (char c in numbers[i])
            {
                int digit = c - '0';
                sum += digit;
                product *= digit;
            }
            if (sum < product)
            {
                Console.Write($"{i} ");
            }
        }
    }
}
